packages/fabrix/src/registry.tsx

```diff
@@ -24,153 +31,260 @@ export type Field = {
+  >(registry: ComponentRegistry<MP>) {
     return new ComponentRegistry({
-      custom: [
-        ...(this.components.custom ?? []),
-        ...(registry.components.custom ?? []),
-      ],
+      custom: {
+        composite: {
+          ...this.props.custom?.composite,
+          ...registry.props.custom?.composite,
+        } as MergeCustomComponentMap<P["custom"], MP["custom"], "composite">,
+        unit: {
+          ...this.props.custom?.unit,
+          ...registry.props.custom?.unit,
+        } as MergeCustomComponentMap<P["custom"], MP["custom"], "unit">,
+      },
       default: {
-        field:
-          registry.components.default?.field ?? this.components.default?.field,
-        form:
-          registry.components.default?.form ?? this.components.default?.form,
-        formField:
-          registry.components.default?.formField ??
-          this.components.default?.formField,
-        table:
-          registry.components.default?.table ?? this.components.default?.table,
+        ...this.props.default,
+        ...registry.props.default,
       },
     });
   }
 
-  getCustom<T extends CustomComponent["type"]>(name: string, type: T) {
-    return this.customComponentMap.get(`${name}:${type}`) as
-      | ComponentFuncByType<T>
-      | undefined;
+  /**
+   * Get the custom component integrated with `FabrixComponent`.
+   *
+   * This method creates a React component that wraps `FabrixComponent` with the custom component.
+   *
+   * The component takes a following props:
+   * * `query`: The GraphQL query string.
+   * * `customProps`: The custom props that the custom component requires.
+   *
+   * @param name The name of the custom composite component.
+   */
+  getFabrixComponent<
+    K extends P["custom"] extends { composite: CompositeComponentMap }
+      ? KeyOf<P["custom"]["composite"]>
+      : never,
+    CP extends P["custom"] extends { composite: CompositeComponentMap }
+      ? P["custom"]["composite"][K]
+      : never,
+  >(name: K) {
+    const componentEntry = this.props.custom?.composite?.[name];
+    if (!componentEntry) {
+      throw new Error(`Component ${name} not found`);
+    }
+
+    return (
+      props: FabrixComponentProps & {
+        customProps: CP extends CompositeComponentEntries<infer P> ? P : never;
+      },
+    ) => (
+      <FabrixCustomComponent
+        {...props}
+        component={{
+          name,
+          entry: componentEntry,
+          customProps: props.customProps,
+        }}
+      >
+        {props.children}
+      </FabrixCustomComponent>
+    );
+  }
+
+  /**
+   * Get the component by name.
+   *
+   * If the component is not found, it will return the default component.
+   *
+   * @param name The name of the component.
+   * @param type The type of the component.
+   */
+  getCustomComponentByNameWithFallback<T extends ComponentEntries["type"]>(
+    name: string | null | undefined,
```

Comment:
Why not just a `string`?
```suggestion
    name: string
```

Reply:
Is it better?
```tsx
  getCustomComponentByNameWithFallback<
    T extends ComponentEntries["type"],
    N extends P["custom"] extends { composite: CompositeComponentMap }
      ? KeyOf<P["custom"]["composite"]>
      : string | null | undefined,
  >(name: N, type: T) {
```

Reply:
https://github.com/fabrix-framework/fabrix/pull/93/commits/0638dbfe054ab831ecb99b43929a9c17c0b9e99b